app/src/main/java/org/openobservatory/ooniprobe/dao/MeasurementDao.java

```diff
@@ -0,0 +1,36 @@
+package org.openobservatory.ooniprobe.dao;
+
+import com.raizlabs.android.dbflow.sql.language.OperatorGroup;
+import com.raizlabs.android.dbflow.sql.language.SQLOperator;
+import com.raizlabs.android.dbflow.sql.language.SQLite;
+
+import org.openobservatory.ooniprobe.model.database.Measurement;
+import org.openobservatory.ooniprobe.model.database.Measurement_Table;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import androidx.annotation.Nullable;
+
+public class MeasurementDao {
+	public static List<Measurement> queryList(@Nullable Integer resultId, @Nullable Integer measurementId, @Nullable Boolean uploaded, @Nullable Boolean failed) {
```

Comment:
i thinked this method to use on all cases we need a measurement list from DB, using params to add where clausole if they are not null.

anyway first 2 params will called as null in some cases now.

Reply:
> anyway first 2 params will called as null in some cases now.

 In this PR? Did I miss it?

Reply:
I agree with @bassosimone I don't see the need to abstract on top of something which already is an abstraction on top of SQL.

Effectively you are using the arguments to this function call to determine how the SQL filtering should happen, yet using directly the ORM semantics is much more expressive. I think that since you have chosen to use a query builder already you should be using that.

Moreover you are calling this method in the same way (`queryList()`) as the query builder method which I think creates some confusion.

I would suggest dropping this method and moving this code directly into the single consume of it (and perhaps simplifying it a bit which it can given that it's only a single use).

Reply:
removed DAO